salt/client/mixins.py

```diff
@@ -240,6 +240,12 @@ def over_print(output):
         event.fire_event(data, tagify('ret', base=tag))
         # if we fired an event, make sure to delete the event object.
         # This will ensure that we call destroy, which will do the 0MQ linger
+        #
+        # The process ends and the zmq linger appears NOT to be respected. As
+        # a result, at time the del event kills the socket before the above
+        # fire_event can finish. There may be a better way to wait for the event
+        # to fire but for now we will insert a small delay.
+        time.sleep(0.25)
```

Comment:
There may be a better way to do this. (There almost certainly is.) I just included this as a placeholder so we can figure out what's happening here. Hopefully the sleep isn't the final answer. :]